nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/AsyncService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nelson Silva <nsilva@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.api;
+
+import java.io.Serializable;
+
+/**
+ * Interface to be implemented by asynchronous services.
+ *
+ * @param <K> type of command id
```

Comment:
AsyncService is supposed to be generic so maybe we should talk about "asynchronous task" and "task identifier" instead of commandId used in bulk service

Reply:
Yes, agree 💯 My main issue was that I had to introduce the `AsyncStatus` so that we could keep bulk operations returning it and not just the commandId, and the method was `getCommandId()` which would require changing `BulkStatus`... unless we go with a more generic `getId()` for the status, wdyt?

Reply:
yes I am fine with a status.getId()

Reply:
👍